RetailCoder.VBE/Inspections/QuickFixes/AssignedByValParameterMakeLocalCopyQuickFix.cs

```diff
@@ -99,11 +102,23 @@ private void ReplaceAssignedByValParameterReferences(IModuleRewriter rewriter, s
 
         private void InsertLocalVariableDeclarationAndAssignment(IModuleRewriter rewriter, string localIdentifier)
         {
-            var content = Tokens.Dim + " " + localIdentifier + " " + Tokens.As + " " + _target.AsTypeName + Environment.NewLine
-                + (_target.AsTypeDeclaration is ClassModuleDeclaration ? Tokens.Set + " " : string.Empty)
-                + localIdentifier + " = " + _target.IdentifierName;
+            var content = Tokens.Dim + " " + localIdentifier + " " + Tokens.As + " " + _target.AsTypeName + Environment.NewLine;
+            string assignmentFormat = "{0} = {1}";
+            if (RequiresSetAssignment(_target))
+            {
+                assignmentFormat = "Set {0} = {1}";
+            }
 
+            content = content + string.Format(assignmentFormat, localIdentifier, _target.IdentifierName);
             rewriter.InsertBefore(((ParserRuleContext)_target.Context.Parent).Stop.TokenIndex + 1, "\r\n" + content);
```

Comment:
Either way, it doesn't matter.  String interpolation still calls `string.Format` behind the scenes:
```
var i = 9;
var s = $"test: {i}";
```
->
```
IL_0000:  nop         
IL_0001:  ldc.i4.s    09 
IL_0003:  stloc.0     // i
IL_0004:  ldstr       "test: {0}"
IL_0009:  ldloc.0     // i
IL_000A:  box         System.Int32
IL_000F:  call        System.String.Format
IL_0014:  stloc.1     // s
IL_0015:  ret
```